p1 batch 2 - 3 more pages in "build tests" created (#492)
OctoMind-dev/mintlify-docs: build-tests/test-setup.mdx

---
title: Test setup
description: "A checklist to setup your test suite quickly."
icon: "wrench"
---

> Test script is just a part of the job setting up a working test suite needs a bit more. We've built Octomind to make your set-up fast and as little painful as possible.

Reviewer: ```suggestion
Test setup is just a part of the job setting up a working test suite. We've built Octomind to make your set-up fast and as painless as possible.
```

## Reaching your app

> 1. **Allowing Octomind bot access your application**

Reviewer: ```suggestion
1. **Allowing Octomind to access your application**
```

   We need access to your application first to test it. In case you have a stringent bot protection, our Octomind bot will have to get whitelisted - it's how Octomind navigates and crawls web pages in order to generate end-to-end tests. Here are all the details explaining how [Octomind bot works](/advanced/octomind-bot).

2. **Testing private apps**

   We are able to test applications that are not publicly accessible thanks to our open-sourced [private location worker](/advanced/private-location). It's open-source so you can review its code before setting it up.

> 3. **Configure exception for avoid rate limiting**

Reviewer: ```suggestion
3. **Configure exception to avoid rate limiting**
```

   We deploy strategies not to trigger rate limiters when creating and running tests like shared auth state, but it's possible we hit it anyways. The easier solution is to [configure an exception to our IP addresses](/advanced/rate-limiting).

## Location

You can decide what proxy servers to use for a specific test target and a specific environment. This is [how to access from different geographies or a private location](/proxy/proxy-settings.mdx).

## You can set up multiple environments

You can configure testing to match specific deployment settings and validate changes across different stages. You'll find `environments` settings it the left project menu.

After creating a project, Octomind provides you with a `default environment` with the url you provided during setup. You can create [multiple environment and set them up separately](/account-view/environments).

<Frame caption="Environment settings in project menu, 03/2025">
  <img
    src="/images/build/environments-settings.png"
    alt="Environment settings in project menu"
  />
</Frame>

## Variables

> Variables are a helpful tool to handle test data and isolate. With Octomind, you can chose from a set of [pre-defined variables, create custom ones or dynamic variables](/advanced/variables). They can be set-up in the left project menu under `environments` and `global variables` tab in each environment.

Reviewer: ```suggestion
Variables are a helpful tool to handle test data and isolate. With Octomind, you can chose from a set of [pre-defined variables, create custom ones or dynamic variables](/advanced/variables). They can be set-up in the left project menu under the `environments` and `global variables` tab in each environment.
```
